lib/Runtime/Language/SourceTextModuleRecord.cpp

```diff
@@ -576,12 +601,27 @@ namespace Js
         {
             this->errorObject = exceptionObject->GetThrownObject(scriptContext);
         }
+        END_ENTER_SCRIPT
 
         if (this->errorObject != nullptr)
         {
             NotifyParentsAsNeeded();
-            return; 
+            return;
+        }
+
+        SetWasDeclarationInitialized();
+        if (childrenModuleSet != nullptr)
```

Comment:
I tried but found that if we try to call ModuleDeclarationInstantiation on the child module set before initializing our exports, the cyclical case asserts because child module tries to resolve an export in parent module before slots are allocated. Maybe we should ensure the slots are allocated before resolving exports?